syntax = "proto3";

package protos;

message ChunkServerLocation {
    string server_hostname = 1;

    uint32 server_port = 2;
}

message ChunkServer {
    ChunkServerLocation location = 1;

    // Amount of disk in megabytes available on the server
    uint32 available_disk_mb = 2;

    // The chunk handles for chunks on this server.
    repeated string stored_chunk_handles = 3;
}

message FileChunk {
    // the version of the chunk
    uint32 version = 1;

    bytes data = 2;
}